src/XPurchase/VirtoCommerce.XPurchase/Schemas/PurchaseSchema.cs

```diff
@@ -643,5 +654,30 @@ private async Task<object> ResolveConnectionAsync(IMediator mediator, IResolveCo
 
             return result;
         }
+
+        private async Task CheckAuthAsync(IResolveFieldContext context, object resource)
+        {
+            var userId = context.GetArgument<string>("userId");
+
+            if (userId == null)
+            {
+                throw new ExecutionError($"argument {nameof(userId)} is null");
+            }
+
+            var signInManager = _signInManagerFactory();
+            var user = await signInManager.UserManager.FindByIdAsync(userId) ?? new ApplicationUser
+            {
+                Id = userId,
+                UserName = "Anonymous",
+            };
+
+            var userPrincipal = await signInManager.ClaimsFactory.CreateAsync(user);
+            var authorizationResult = await _authorizationService.AuthorizeAsync(userPrincipal, resource, new CanAccessCartAuthorizationRequirement());
+
+            if (!authorizationResult.Succeeded)
```

Comment:
Please check how to returns 401.
https://github.com/VirtoCommerce/vc-module-experience-api/blob/dev/src/XProfile/VirtoCommerce.XProfile/Schemas/OrganizationType.cs#L21

Reply:
Returning 401 is possible only by overring GraphQL Http server, parsing error messages and integrating our imperative auth checks into their auth system: https://github.com/graphql-dotnet/authorization/issues/1
I think it does not worth it, let's leave it as is.